.pre-commit-config.yaml

```diff
@@ -1 +1,46 @@
-repos: []
+# To enable this pre-commit hook run:
+# `pip install pre-commit` or `brew install pre-commit`
+# Then run `pre-commit install`
+
+# Learn more about this config here: https://pre-commit.com
+
+repos:
+  - repo: https://github.com/pre-commit/pre-commit-hooks
+    rev: v4.5.0
+    hooks:
+      - id: check-toml
+      - id: check-yaml
+      - id: detect-private-key
+      - id: end-of-file-fixer
+      - id: mixed-line-ending
+      - id: requirements-txt-fixer
+      #- id: name-tests-test
+      #  args:
+      #    - --django
+      #- id: trailing-whitespace
+      #  args:
+      #    - --markdown-linebreak-ext=md
+
+  - repo: https://github.com/astral-sh/ruff-pre-commit
+    rev: v0.1.6
+    hooks:
+      - id: ruff
+
+  #- repo: https://github.com/psf/black
+  #  rev: 23.11.0
+  #  hooks:
+  #    - id: black  # See pyproject.toml for args
```

Comment:
See #257 -- We want these lines but adding them causes too many changes so pull requests are not reviewed.